File: TrimList.jl
#!julia.exe
# AUTHOR: Robert DiCicco
# DATE: 2022-08-31
# Challenge #180 Trim List ( Julia )

arr = [[1,4,2,3,5],  [9,0,6,2,3,8,5]]
nums = [3,4]
count = 1
while count < 3
  results = []
  val = arr[count]
  println("Input: @n = ",val," and \$i = ",nums[count])
  chcnt = 1
  for i in val
    if i > nums[count]
      push!(results,i)
    end
  end
  global count += 1
  println("Output: ",results)
  println("")
end

#-----------------------------------------
# SAMPLE OUTPUT 
# PS G:\Projects\Perl\Challenges> julia .\TrimList.jl
# Input: @n = [1, 4, 2, 3, 5] and $i = 3
# Output: Any[4, 5]

# Input: @n = [9, 0, 6, 2, 3, 8, 5] and $i = 4
# Output: Any[9, 6, 8, 5]

File: Lua/TrimList.lua
--[[
AUTHOR: Robert DiCicco
DATE: 2022-09-01
Challenge #180 Trim List ( Lua )
--]]

local write = io.write

function printInputLine(a,n)
	write("Input: ( ")
	for _, x in ipairs(a) do
		write(x,' ')
	end
	write(")")
	write(" and $i = ",n)
	write('\n')
end


function process(a,n)
	write("Output: ( ")
	for _, x in ipairs(a) do
		if x > n then
			write(x,' ')
		end
	end
	write(") ")
	write('\n\n')
end

arr = { 
	{9,0,6,2,3,8,5},
	{1,4,2,3,5}
}

mynums = {4,3}

function main()
	for idx = 1, 2 do
		printInputLine(arr[idx],mynums[idx])
		process(arr[idx],mynums[idx])
	end
end

main()
